Sources/System/FilePath/FilePathTemp.swift

```diff
@@ -0,0 +1,281 @@
+  }
+}
+
+internal func forEachFile(
+  at path: FilePath,
+  _ body: (WIN32_FIND_DATAW) throws -> ()
+) rethrows {
+  let searchPath = path.appending("\\*")
+
+  try searchPath.withPlatformString { szPath in
+    var findData = WIN32_FIND_DATAW()
+    let hFind = FindFirstFileW(szPath, &findData)
+    if hFind == INVALID_HANDLE_VALUE {
+      throw Errno(windowsError: GetLastError())
+    }
+    defer {
+      FindClose(hFind)
+    }
+
+    repeat {
+      // Skip . and ..
+      if findData.cFileName.0 == 46
+           && (findData.cFileName.1 == 0
+                 || (findData.cFileName.1 == 46
+                       && findData.cFileName.2 == 0)) {
+        continue
+      }
+
+      try body(findData)
+    } while FindNextFileW(hFind, &findData)
+  }
+}
+
+internal func recursiveRemove(at path: FilePath) throws {
+  // First, deal with subdirectories
+  try forEachFile(at: path) { findData in
+    if (findData.dwFileAttributes & DWORD(FILE_ATTRIBUTE_DIRECTORY)) != 0 {
+      let name = withUnsafeBytes(of: findData.cFileName) {
+        return SystemString(platformString: $0.assumingMemoryBound(
+                              to: CInterop.PlatformChar.self).baseAddress!)
+      }
+      let component = FilePath.Component(name)!
+      let subpath = path.appending(component)
+
+      try recursiveRemove(at: subpath)
+    }
+  }
+
+  // Now delete everything else
+  try forEachFile(at: path) { findData in
+    let name = withUnsafeBytes(of: findData.cFileName) {
+      return SystemString(platformString: $0.assumingMemoryBound(
+                            to: CInterop.PlatformChar.self).baseAddress!)
+    }
+    let component = FilePath.Component(name)!
+    let subpath = path.appending(component)
+
+    if (findData.dwFileAttributes & DWORD(FILE_ATTRIBUTE_DIRECTORY)) == 0 {
+      try subpath.withPlatformString {
+        if !DeleteFileW($0) {
+          throw Errno(windowsError: GetLastError())
+        }
+      }
+    }
+  }
+
+  // Finally, delete the parent
+  try path.withPlatformString {
+    if !RemoveDirectoryW($0) {
+      throw Errno(windowsError: GetLastError())
+    }
+  }
+}
+
+#else
+internal func getTemporaryDirectory() throws -> FilePath {
+  #if SYSTEM_PACKAGE_DARWIN
+  var capacity = 1024
+  while true {
+    let path: FilePath? = withUnsafeTemporaryAllocation(
+      of: CInterop.PlatformChar.self,
+      capacity: capacity) { buffer in
+      let len = system_confstr(SYSTEM_CS_DARWIN_USER_TEMP_DIR,
+                               buffer.baseAddress!,
+                               buffer.count)
+      if len == 0 {
+        // Fall back to "/tmp" if we can't read the temp directory
+        return "/tmp"
+      }
+      // If it was truncated, increase capaciy and try again
+      if len > buffer.count {
+        capacity = len
+        return nil
+      }
+      return FilePath(SystemString(platformString: buffer.baseAddress!))
+    }
+    if let path = path {
+      return path
+    }
```

Comment:
This seems _far_ too much logic for swift-system. This package is not supposed to be the place for building cross-platform abstractions.